### DISCUSSION TITLE
GitHub SSO Re-Authentication Timeout

### DISCUSSION THREAD

**Discussion by rnhurt** [community/community#50334] [Enterprise]
### Select Topic Area

Question

### Body

We recently started using SSO to authenticate to GitHub and everything seems to be working fine.  The one thing that is bothering me is the re-authentication timeout.  I'm not sure how long the "session" is supposed to last but it feels like I'm re-authenticating multiple times per day.  

It's not a huge problem, but it is a bit jarring when the browser refreshes while your trying to update an Issue or modify your Project.  Most other systems don't really re-auth that often so I'm not really sure why GitHub is so needy.  Hell, before we moved to SSO I didn't re-auth at all unless I had to do something "risky" (create a PAT, delete a repo, etc.) Why does SSO require so much re-authentication?

### Questions
1. How long does a GitHub SSO auth token last?
2. Is there a way to lengthen the time?

**Comment by Anmol-Baranwal** [comment]
The default timeout for GitHub SSO authentication tokens is six hours. This means that after six hours of inactivity, you will be prompted to re-authenticate.

However, GitHub does provide a way to adjust the session timeout. This can be done by the organization owner or administrator in the organization's SAML settings. Here's how:

1. Navigate to your organization's settings in GitHub.
2. Click on the "Security & analysis" tab.
3. Click on "Single sign-on" in the left-hand navigation menu.
4. Scroll down to the "Session duration" section.
5. Adjust the session duration to your preferred length of time.
6. Click "Save" to apply the changes.

> Note that the maximum session duration is 12 hours, so you won't be able to set it to last longer than that.

Keep in mind that a longer session duration can increase the risk of unauthorized access if your device is lost or stolen, so it's important to balance convenience with security when adjusting this setting.

**Reply by rnhurt** [reply]
That sounds like exactly what I was looking for.  Unfortunately I'm not able to navigate to the page you're talking about.  I originally configured the SAML so I should have the permissions I need.  

When I go to the organization's settings in GitHub I can see sections labeled "Authentication security" & "Code security & analysis" but nothing labeled "Security & analysis".  I clicked through both sections and couldn't find a "Single sign-on" page either.  We're on the "Enterprise" plan so maybe that changes the menu/options a bit or something.  Either way, I was not able to find a "Session duration" setting anywhere.  :shrug:

**Reply by Anmol-Baranwal** [reply]
To locate the "Session duration" setting for your Enterprise plan, try this:

In the Security & analysis menu, click on "Authentication."
Scroll down to the "Session duration" section, where you can adjust the length of time for SAML sessions.

If you're still unable to locate this setting, you may want to consider waiting for others to respond, as they could potentially provide additional insights or a better solution.

**Reply by Anmol-Baranwal** [reply]
@rnhurt 
I just wanted to check in and see if you have any remaining questions or concerns about my answer. If not, would you please mark my response as accepted?

It would be greatly appreciated and it will also help other users who might be looking for a solution to a similar issue.

**Reply by rnhurt** [reply]
@Anmol-Baranwal I'm sorry, but I still can't find the menus you are describing.  My "Code security & analysis" section doesn't have a "Authentication" item to click on or select.  And I still can't find a "Session duration" section anywhere in any of my preferences.  🤷 

![image](https://user-images.githubusercontent.com/24466/226176492-2484072d-806c-4e2a-9784-39cd2e27e08a.png)

**Reply by Anmol-Baranwal** [reply]
Ah, no worries. Maybe the source of my information is wrong. Let others answer this for a satisfactory response.

**Reply by reeseglasscock** [reply]
Please note that currently the only method available for setting the SAML session duration is setting the `SessionNotOnOrAfter`: https://docs.github.com/en/enterprise-cloud@latest/admin/identity-and-access-management/using-saml-for-enterprise-iam/saml-configuration-reference#session-duration-and-timeout

There is currently no UI configuration available on GitHub.com to adjust SAML session timeouts.
